bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/internal/community/CommunityTransformationAddonHandler.java

```diff
@@ -151,7 +150,8 @@ private PersistedTransformation addTransformationFromYAML(String id, String yaml
 
     private PersistedTransformation addTransformationFromJSON(String id, String json) {
         try {
-            PersistedTransformation transformation = gson.fromJson(json, PersistedTransformation.class);
+            PersistedTransformation transformation = Objects
+                    .requireNonNull(gson.fromJson(json, PersistedTransformation.class));
```

Comment:
Looking at the docs of `fromJson` it could return `null`. `json` does probably contain a valid Json object at this time, so it will likely not happen. Otherwise we would have would have seen some issues here....

So "fail fast" is the correct approach here.

Is my understanding correct?

Reply:
Yes, you're correct. I thought that if we had an issue the way the code is written would already generate an NPE

Reply:
For completeness: `fromJson` returns null if json is null, so this should simply be avoided.